cli/core/test/e2e/config/config_test.go

```diff
@@ -0,0 +1,59 @@
+// Package config_e2e_test provides config command specific E2E test cases
+package config_e2e_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/cluster-api/util"
+
+	"github.com/vmware-tanzu/tanzu-framework/cli/core/test/e2e/framework"
+)
+
+var _ = framework.CLICoreDescribe("[Tests:E2E][Feature:Command-Config]", func() {
+	var (
+		tf *framework.Framework
+	)
+	BeforeEach(func() {
+		tf = framework.NewFramework()
+	})
+	Context("config feature flag operations", func() {
+		When("new config flag set with value", func() {
+			It("should set flag and unset flag successfully", func() {
+				randomFlagName := RandomNameWithE2ETestString()
+				randomFeatureFlagPath := "features.global." + randomFlagName
+				flagVal := "true"
+				err := tf.ConfigSetFeature(randomFeatureFlagPath, flagVal)
+				Expect(err).To(BeNil())
+
+				val, err := tf.ConfigFeatureFlagExists(randomFlagName, flagVal)
+				Expect(err).To(BeNil())
+				Expect(val).To(BeTrue())
+
+				err = tf.ConfigUnsetFeature(randomFeatureFlagPath)
+				Expect(err).To(BeNil())
+
+				val, err = tf.ConfigFeatureFlagNotExists(randomFlagName)
+				Expect(err).To(BeNil())
+				Expect(val).To(BeTrue())
+			})
+		})
+		When("config init called", func() {
+			It("should initialize configuration successfully", func() {
+				err := tf.ConfigInit()
+				Expect(err).To(BeNil())
+			})
+		})
+
+		// TODO: test list and delete use cases as part of context or login commands
+		When("config server list called", func() {
+			It("should list all available servers", func() {
+				err := tf.ConfigServerList()
+				Expect(err).To(BeNil())
+			})
+		})
```

Comment:
Let's not add a partial test for server list here. I am not seeing much meaning of this test here considering we are not adding or deleting any servers for the tests. It might be good to add tests for this along with other scenarios in a separate test file.

Reply:
I have removed tests for now, i will add servers specific tests in separate file.